### DISCUSSION TITLE
Remove the macro detections

### DISCUSSION THREAD

**Discussion by ySergi** [J-Tanzanite/Little-Anti-Cheat#195] [Ideas]
It's rubbish, I've already been banned on two servers for using autoshoot. For accessibility reasons, I use a Razer macro that converts semi-automatic weapons to automatic and I've been banned for using it. 

From what I have read, the ban for this detection is 60 minutes maximum. Thanks for not making it permanent at least as it would be horrible. 

Please remove this feature as it is annoying and counterproductive for those of us who play with macros. I know this wouldn't make the current servers remove this detection, but at least more servers in the future wouldn't have this annoying detection.

**Comment by J-Tanzanite** [comment]
I understand that macros has legit use cases, but unfortunately, this feature exists because there also are legit reasons why some server owners don't want people to use macros.

Most server owners (depending a bit on the game) do not care about Macros, and so it is disabled by default.
If enabled, it defaults to first warning users who use Macros to stop, then to kick (not ban).
Macros also only work on servers with 60 ticks per second or more, so games like L4D1&2 are ignored.

I have mainly two concerns with removing macros as a feature:
1. Macros have a chance of causing false positives for some cheat detections, it's rare, but it _can_ happen.
2. Server owners who don't want Macros will _always_ have some plugin to default to for detecting Macros, and other plugin authors may decide to lower the threshold for Clicks Per Second, and/or perma ban macro users; removing this feature from Lilac may be counter productive.

Lilac defaults to detecting Macros who act 20 times a second or more, so a bandaid is to lower your Macros to only act 15-18 times a second.

All that said, it's a shame that it escaped my mind that perhaps, there should be a warning displayed to server owners when they enable Macro detection that it may affect accessibility.
